src/libbson/src/bson/bson-vector.h

```diff
@@ -0,0 +1,591 @@
+BSON_EXPORT (bool)
+bson_vector_int8_view_init (bson_vector_int8_view_t *view_out, uint8_t *binary_data, uint32_t binary_data_len);
+
+BSON_EXPORT (bool)
+bson_vector_int8_const_view_init (bson_vector_int8_const_view_t *view_out,
+                                  const uint8_t *binary_data,
+                                  uint32_t binary_data_len);
+
+BSON_EXPORT (bool)
+bson_vector_float32_view_init (bson_vector_float32_view_t *view_out, uint8_t *binary_data, uint32_t binary_data_len);
+
+
+BSON_EXPORT (bool)
+bson_vector_float32_const_view_init (bson_vector_float32_const_view_t *view_out,
+                                     const uint8_t *binary_data,
+                                     uint32_t binary_data_len);
+
+BSON_EXPORT (bool)
+bson_vector_packed_bits_view_init (bson_vector_packed_bits_view_t *view_out,
+                                   uint8_t *binary_data,
+                                   uint32_t binary_data_len);
+
+BSON_EXPORT (bool)
+bson_vector_packed_bits_const_view_init (bson_vector_packed_bits_const_view_t *view_out,
+                                         const uint8_t *binary_data,
+                                         uint32_t binary_data_len);
+
+
+BSON_EXPORT (bool)
+bson_vector_int8_view_from_iter (bson_vector_int8_view_t *view_out, bson_iter_t *iter);
+
+BSON_EXPORT (bool)
+bson_vector_int8_const_view_from_iter (bson_vector_int8_const_view_t *view_out, const bson_iter_t *iter);
+
+BSON_EXPORT (bool)
+bson_vector_float32_view_from_iter (bson_vector_float32_view_t *view_out, bson_iter_t *iter);
+
+BSON_EXPORT (bool)
+bson_vector_float32_const_view_from_iter (bson_vector_float32_const_view_t *view_out, const bson_iter_t *iter);
+
+BSON_EXPORT (bool)
+bson_vector_packed_bits_view_from_iter (bson_vector_packed_bits_view_t *view_out, bson_iter_t *iter);
+
+BSON_EXPORT (bool)
+bson_vector_packed_bits_const_view_from_iter (bson_vector_packed_bits_const_view_t *view_out, const bson_iter_t *iter);
+
+
+BSON_EXPORT (bool)
+bson_array_builder_append_vector_int8_elements (struct _bson_array_builder_t *builder,
+                                                bson_vector_int8_const_view_t view);
+
+BSON_EXPORT (bool)
+bson_array_builder_append_vector_float32_elements (struct _bson_array_builder_t *builder,
+                                                   bson_vector_float32_const_view_t view);
+
+BSON_EXPORT (bool)
+bson_array_builder_append_vector_packed_bits_elements (struct _bson_array_builder_t *builder,
+                                                       bson_vector_packed_bits_const_view_t view);
+
+BSON_EXPORT (bool)
+bson_array_builder_append_vector_elements (struct _bson_array_builder_t *builder, const bson_iter_t *iter);
+
+
+BSON_EXPORT (bool)
+bson_append_vector_int8 (
+   bson_t *bson, const char *key, int key_length, size_t element_count, bson_vector_int8_view_t *view_out);
+
+#define BSON_APPEND_VECTOR_INT8(b, key, count, view) bson_append_vector_int8 (b, key, (int) strlen (key), count, view)
```

Comment:
```suggestion
BSON_EXPORT (bool)
bson_append_vector_int8_uninit (
   bson_t *bson, const char *key, int key_length, size_t element_count, bson_vector_int8_view_t *view_out);

#define BSON_APPEND_VECTOR_INT8_UNINIT(b, key, count, view) \
   bson_append_vector_int8_uninit (b, key, (int) strlen (key), count, view)
```
Consider suffixing `_uninit` and `_UNINIT` to this function (and the other two functinons below for float32 and packed_bits) for consistency with calls to `bson_append_binary_uninit` and better communicate our expectation for users to fill in the uninitialized bytes.